server/src/main/java/io/crate/types/DataType.java

```diff
@@ -203,4 +211,12 @@ public void writeTo(StreamOutput out) throws IOException {
     public String toString() {
         return getName();
     }
+
+
+    public ColumnType<Expression> toColumnType(ColumnPolicy columnPolicy,
+                                               @Nullable Supplier<List<ColumnDefinition<Expression>>> convertChildColumn) {
```

Comment:
I'm not too happy about the `convertChildColumn` argument - it is required because some information is kept in the `Reference` implementation - e.g. for geo shapes we've the precision in the `GeoReference`.  (See `extractColumnDefinitions` in `MetadataToASTNodeResolver`


To me this is a bit of a sign that this information should become part of the `DataType` as well (but we can follow up on that later)